src/updates/node/index.ts

```diff
@@ -0,0 +1,75 @@
+import { exec } from 'child_process';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+import * as semver from 'semver';
+import got from 'got';
+
+function getVersion(): string | undefined {
+	let version = '';
+	try {
+		exec('node --version', function (err, stdout) {
+			if (err) {
+				throw err;
+			}
+			version = stdout;
+		});
+	} catch (e) {
+		return;
+	}
+
+	return version;
+}
+
+export function checkInstalledVersion(): string | undefined {
+	const version = getVersion();
+
+	if (version && !semver.satisfies(version, '>=10.13')) {
+		throw new Error('Titanium requires node 10.13 or greater.');
+	}
+
+	return version;
+}
+
+export async function checkLatestVersion(sdkPath?: string): Promise<string | undefined> {
+
+	let supportedVersions = '10.x || 12.x';
```

Comment:
I am unsure where I should put this still if I should pass it from the editors or any better options, I would love your input